### DISCUSSION TITLE
FIP-47 Allow FIO Block Producers to update their information

### DISCUSSION THREAD

**Discussion by ericbutz** [fioprotocol/fips#317] [FIPs]
Tracks discussions about [FIP-47](https://github.com/fioprotocol/fips/blob/master/fip-0047.md)

This FIP proposes to extend FIO Protocol functionality to enable FIO Block Producers to use [regproducer](https://developers.fioprotocol.io/pages/api/fio-api/#options-regproducer) to update their producer information.

**Comment by RalfWeinand** [comment]
Hi @ericbutz , this proposal looks good to me. 
Some minor feedback / questions:
---
Section "Rationale": I thought the reason for this change is that it is not possible to update the records of a producer without first unregistering. If a producer wants to switch between producer nodes with a different key for failover this is an extra step and takes longer for failover to become active.

Question about allowing to update "fio_address" of the producer: I have no concern with that, just wondering, is there a particular use case for this. 

Regarding "Exception handling conditions": The "fio_address" isn't listed in the sentence "If the producer is active, assert the url, public_key or location are different and can be updated". Any reason for that ?

**Reply by ericbutz** [reply]
Definitely, the main use case is swapping out producer keys without having to unregister. We are going to allow updating FIO handle as well. Probably rare, but maybe a group rebrands or wants to use a different handle... since votes are linked to account it will not impact things.

**Comment by ericbutz** [comment]
Based on feedback, looks like we are good to go for moving FIP-47 to "accepted." 

https://github.com/fioprotocol/fips/blob/master/fip-0047.md
